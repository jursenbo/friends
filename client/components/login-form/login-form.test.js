import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import LoginForm from './'

jest.mock('../../utils', () => {
	return {
		validateEmail: jest.fn(un => un === 'testUsername'),
	}
})

jest.mock('lodash', () => {
	return {
		debounce: jest.fn(func => func),
	}
})

describe('LoginForm snapshots', () => {
	test('renders form', () => {
		const component = renderer.create(<LoginForm submitLoginForm={() => {}} />)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('renders form with message', () => {
		const component = renderer.create(
			<LoginForm submitLoginForm={() => {}} message="message" />
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('renders form with error', () => {
		const component = renderer.create(
			<LoginForm submitLoginForm={() => {}} error="error" />
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})

describe('LoginForm methods', () => {
	test('componentDidUpdate', () => {
		const loginForm = shallow(<LoginForm submitLoginForm={() => {}} />)
		expect(loginForm.state('error')).toBeNull()
		expect(loginForm.state('message')).toBeNull()
		loginForm.setProps({ message: 'testMessage', error: 'testError' })
		loginForm.instance().componentDidUpdate({})
		expect(loginForm.state('error')).toBe('testError')
		expect(loginForm.state('message')).toBe('testMessage')
	})

	test('onUsernameChange', () => {
		const username = 'testUsername'
		const loginFormInstance = shallow(
			<LoginForm submitLoginForm={() => {}} />
		).instance()
		expect(loginFormInstance.state.username).toBeNull()
		loginFormInstance.username = { value: username }
		loginFormInstance.onUsernameChange()
		expect(loginFormInstance.state.username).toBe(username)
		loginFormInstance.username = { value: 'abc' }
		loginFormInstance.onUsernameChange()
		expect(loginFormInstance.state.username).toBeNull()
		expect(loginFormInstance.state.error).toBe(
			'Please enter a valid e-mail address.'
		)
	})
})
