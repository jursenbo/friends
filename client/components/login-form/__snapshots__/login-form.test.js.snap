// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshots renders form 1`] = `
<section
  className="login-form card card--500px"
>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputName"
      >
        E-mail
      </label>
      <input
        className="form-control"
        id="inputName"
        onChange={[Function]}
        placeholder="E-mail"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;

exports[`LoginForm snapshots renders form with error 1`] = `
<section
  className="login-form card card--500px"
>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputName"
      >
        E-mail
      </label>
      <input
        className="form-control"
        id="inputName"
        onChange={[Function]}
        placeholder="E-mail"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;

exports[`LoginForm snapshots renders form with message 1`] = `
<section
  className="login-form card card--500px"
>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputName"
      >
        E-mail
      </label>
      <input
        className="form-control"
        id="inputName"
        onChange={[Function]}
        placeholder="E-mail"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
